homeassistant/components/overkiz/alarm_control_panel.py

```diff
@@ -0,0 +1,298 @@
+        alarm_trigger=OverkizCommand.ALARM_ON,
+    ),
+    # StatefulAlarmController
+    OverkizAlarmDescription(
+        key=UIWidget.STATEFUL_ALARM_CONTROLLER,
+        supported_features=(
+            SUPPORT_ALARM_ARM_AWAY | SUPPORT_ALARM_ARM_HOME | SUPPORT_ALARM_ARM_NIGHT
+        ),
+        fn_state=_state_stateful_alarm_controller,
+        alarm_disarm=OverkizCommand.DISARM,
+        alarm_arm_home=OverkizCommand.ALARM_ZONE_ON,
+        alarm_arm_home_args=[OverkizCommandParam.A],
+        alarm_arm_night=OverkizCommand.ALARM_ZONE_ON,
+        alarm_arm_night_args=[OverkizCommandParam.A, OverkizCommandParam.B],
+        alarm_arm_away=OverkizCommand.ALARM_ZONE_ON,
+        alarm_arm_away_args=[
+            OverkizCommandParam.A,
+            OverkizCommandParam.B,
+            OverkizCommandParam.C,
+        ],
+    ),
+    # MyFoxAlarmController
+    OverkizAlarmDescription(
+        key=UIWidget.MY_FOX_ALARM_CONTROLLER,
+        supported_features=SUPPORT_ALARM_ARM_AWAY | SUPPORT_ALARM_ARM_NIGHT,
+        fn_state=_state_myfox_alarm_controller,
+        alarm_disarm=OverkizCommand.DISARM,
+        alarm_arm_night=OverkizCommand.PARTIAL,
+        alarm_arm_away=OverkizCommand.ARM,
+    ),
+    # AlarmPanelController
+    OverkizAlarmDescription(
+        key=UIWidget.ALARM_PANEL_CONTROLLER,
+        supported_features=(
+            SUPPORT_ALARM_ARM_AWAY | SUPPORT_ALARM_ARM_HOME | SUPPORT_ALARM_ARM_NIGHT
+        ),
+        fn_state=_state_alarm_panel_controller,
+        alarm_disarm=OverkizCommand.DISARM,
+        alarm_arm_home=OverkizCommand.ARM_PARTIAL_DAY,
+        alarm_arm_night=OverkizCommand.ARM_PARTIAL_NIGHT,
+        alarm_arm_away=OverkizCommand.ARM,
+    ),
+]
+
+SUPPORTED_DEVICES = {description.key: description for description in ALARM_DESCRIPTIONS}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Overkiz alarm control panel from a config entry."""
+    data: HomeAssistantOverkizData = hass.data[DOMAIN][entry.entry_id]
+    entities: list[OverkizAlarmControlPanel] = []
+
+    for device in data.platforms[Platform.ALARM_CONTROL_PANEL]:
+        if description := SUPPORTED_DEVICES.get(device.widget) or SUPPORTED_DEVICES.get(
+            device.ui_class
+        ):
+            entities.append(
+                OverkizAlarmControlPanel(
+                    device.device_url,
+                    data.coordinator,
+                    description,
+                )
+            )
+
+    async_add_entities(entities)
+
+
+class OverkizAlarmControlPanel(OverkizDescriptiveEntity, AlarmControlPanelEntity):
+    """Representation of an Overkiz Alarm Control Panel."""
+
+    entity_description: OverkizAlarmDescription
+
+    @property
+    def supported_features(self) -> int:
+        """Flag supported features."""
+        return self.entity_description.supported_features
```

Comment:
Isn't this the default to read the entity description ?

Reply:
Not that I know. It seems that you still need to set `_attr_supported_features`. Apparently this is not supported in EntityDescription (which makes sense, since we needed to add this field ourselves).

Reply:
Could/  should it set _attr_supported_features in __init__ ?

Reply:
2a0fe445519f2f4b27cd9efd0c661dcab4423295